pyspark/pyproject.toml

```diff
@@ -0,0 +1,40 @@
+[tool.poetry]
+name = "graphar-pyspark"
+version = "0.0.1"
+description = "PySpark bindings to GraphAr scala spark package"
+authors = ["Semyon Sinchenko ssinchenko@pm.me"]
+readme = "README.rst"
+packages = [{include = "graphar_pyspark"}]
+
+[tool.poetry.dependencies]
+python = "^3.9"
+pyyaml = "*"
+
+[tool.poetry.group.spark]
+optional = true
+
+[tool.poetry.group.spark.dependencies]
+pyspark = "^3.2.2" # TODO: relax requirement when scala part will be available for multiple spark versions
+
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+ruff = "*"
+
+[tool.poetry.group.jupyter]
+optional = true
+
+[tool.poetry.group.jupyter.dependencies]
+jupyterlab = "*"
```

Comment:
Even though this is (IIUC) an optional dependency, I'd suggest removing this before merging. This is a huge dependency which is quite case-specific, and the users of this library probably don't need it.